File: garage.py
class Garage:

    # Membuat data atribut jenis motor, kapasitas tangki
    __nama_garasi = ""
    __luas_garasi = ""
    __daftar_kendaraan = ""

    # Constructor
    def __init__(self, nama_garasi = "", luas_garasi = "", daftar_kendaraan = ""): # Membuat constructor dengan parameter
        # menginisialisasi masing masing atribut dengan nilai dari parameter nama atribut sesuai namanya
        self.__nama_garasi = nama_garasi
        self.__luas_garasi = luas_garasi
        self.__daftar_kendaraan = daftar_kendaraan
        # self.__car = car # Composite list of object
        # self.__motorcycle = motorcycle # Composite list of object

    # Getter dan Setter

    # Getter    
    def getNama_Garasi(self):
        return self.__nama_garasi
    
    def getLuas_Garasi(self):
        return self.__luas_garasi
    
    def getDaftar_Kendaraan(self):
        return self.__daftar_kendaraan
    
    # def getCar(self):
    #     return self.__car
    
    # def getMotorcycle(self):
    #     return self.__motorcycle

    
    # Setter
    def setNama_Garasi(self, nama_garasi):
        self.__nama_garasi = nama_garasi

    def setLuas_Garasi(self, luas_garasi):
        self.__luas_garasi = luas_garasi

    def setDaftar_Kendaraan(self, daftar_kendaraan):
        self.__daftar_kendaraan = daftar_kendaraan

    # def setCar(self, car):
    #     self.__car = car

    # def setMotorcycle(self, motorcycle):
    #     self.__motorcycle = motorcycle

File: C++/Garage.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>

// Deklarasi namespace
using namespace std;

class Garage
{
private:
    /* data */
    string nama_garasi;
    string luas_garasi;
    string daftar_kendaraan;

public:
    /* Constructor */
    Garage() /* Membuat Constructor Default tanpa parameter */
    {
        this->nama_garasi = "";
        this->luas_garasi = "";
        this->daftar_kendaraan = "";
    }

    Garage(string nama_garasi, string luas_garasi, string daftar_kendaraan)
    {
        this->nama_garasi = nama_garasi;
        this->luas_garasi = luas_garasi;
        this->daftar_kendaraan = daftar_kendaraan;
    }
    /* Getter dan Setter*/

    /* Getter*/
    string getNamaGarasi()
    {
        return this->nama_garasi;
    }

    string getLuasGarasi()
    {
        return this->luas_garasi;
    }

    string getDaftarkendaraan()
    {
        return this->daftar_kendaraan;
    }

    /* Setter */
    void setNamaGarasi(string nama_garasi)
    {
        this->nama_garasi = nama_garasi;
    }

    void setLuasGarasi(string luas_garasi)
    {
        this->luas_garasi = luas_garasi;
    }

    void setDaftarkendaraan(string daftar_kendaraan)
    {
        this->daftar_kendaraan = daftar_kendaraan;
    }

    /* Destructor */
    /* Membuat Destructor untuk menghapus semua objek yang telah dibuat*/
    ~Garage()
    {
    }
};
